### PR TITLE
Add support for generating README.md data

### PR DESCRIPTION
**Pull request by amckinney**
This adds support for generating a `README.md` file which will be consumed by Fern's remote generation service. Right now, the `README.md` file will only be produced when the configured output has a publish configuration (i.e. GitHub or PyPi), which is identified by a non-`None` `ProjectConfig`.

### CODE REVIEW

**Comment by amckinney** [comment]
This shouldn't be merged until the remote generation server is prepared to act upon the generated `readme.json` file.

**Comment by amckinney** [comment]
I'll need to take a closer look at the snapshot and/or `seed` fixtures because I can see the `readme.json` file being produced for a variety of snapshots in the temporary output directory (hence the example shown in the summary), but all of the tests continue to pass without any seed testdata updates - something seems off.

**Comment by amckinney** [comment]
Ended up making some minor edits to the snapshot testing setup so that it recognizes the `readme.json` file in the cases where in the cases where it's applicable. There changes will be irrelevant once we fully move to `seed`, but we'll need support for custom configurations before we can do so.

File: src/fern_python/generators/sdk/client_generator/root_client_generator.py [hunk 1]
```diff
@@ -108,6 +112,20 @@ def generate(self, source_file: SourceFile) -> None:
...
+            instantiation=f"""```python
+from {module_path} import {self._class_name}
+
+client = {self._class_name}({self._constructor_info.instantiation})
+```""",
+            async_instantiation=f"""```python
+from {module_path} import {self._async_class_name}
+
+client = {self._async_class_name}({self._constructor_info.instantiation})
+```""",
```
Comment (dsinghvi) [inline src/fern_python/generators/sdk/client_generator/root_client_generator.py:126]: nit: can we move lines 117-126 (the mdx in its own helper function)

File: src/fern_python/generators/sdk/core_utilities/client_wrapper_generator.py [hunk 2]
```diff
@@ -385,6 +387,7 @@ def _get_constructor_info(self) -> ConstructorInfo:
                     header_prefix=header_auth_scheme.prefix,
                 )
             )
+            instantiation = f'{header_auth_scheme.name.name.snake_case.safe_name}=<"YOUR_{header_auth_scheme.name.name.screaming_snake_case.safe_name}">'
```
Comment (dsinghvi) [inline src/fern_python/generators/sdk/core_utilities/client_wrapper_generator.py:390]: @amckinney can we add an instantiation for global headers (line 365)

File: src/fern_python/generators/sdk/core_utilities/client_wrapper_generator.py [hunk 3]
```diff
@@ -570,6 +585,11 @@ def _get_auth_scheme_header_constructor_parameter_name(self, header: ir_types.He
     def _get_auth_scheme_header_private_member_name(self, header: ir_types.HeaderAuthScheme) -> str:
         return header.name.name.snake_case.unsafe_name
 
+    def _get_environment_instantiation(self, ir: ir_types.IntermediateRepresentation) -> Optional[str]:
+        if ir.environments is None or ir.environments.default_environment is None:
+            return None
+        return f'{ClientWrapperGenerator.ENVIRONMENT_PARAMETER_NAME}=<"YOUR_ENVIRONMENT">'
```
Comment (dsinghvi) [inline src/fern_python/generators/sdk/core_utilities/client_wrapper_generator.py:591]: if default environment is none we should just not show anything, so this return should be in the if statement on 590

Comment (dsinghvi) [reply src/fern_python/generators/sdk/core_utilities/client_wrapper_generator.py:591]: `return f'{ClientWrapperGenerator.ENVIRONMENT_PARAMETER_NAME}=<"YOUR_ENVIRONMENT">'` we should default to the first environment enum value

Comment (amckinney) [reply src/fern_python/generators/sdk/core_utilities/client_wrapper_generator.py:591]: Discussed offline - that's what the behavior is already.

**Pull request merged**